resources/assets/coffee/react/beatmapset-page/header.coffee

```diff
@@ -28,34 +28,123 @@ class BeatmapsetPage.Header extends React.Component
     return true
 
   render: ->
-    div className: 'osu-layout__row osu-layout__row--page-compact',
+    dateFormat = 'MMM D, YYYY'
+
+    div className: 'osu-layout__row osu-layout__row--beatmapset-header',
+      div className: 'header-tabs',
+        for mode in BeatmapHelper.modes
+          continue if _.isEmpty @props.beatmapList[mode]
+
+          el BeatmapsetPage.HeaderTab,
+            key: mode
+            playmode: mode
+            currentBeatmapId: @props.currentBeatmap.id
+            newBeatmapId: _.last @props.beatmapList[mode]
+            currentPlaymode: @props.currentBeatmap.mode
+
       div
-        className: 'beatmapset-header',
-        onClick: @onClick,
+        className: 'beatmapset-header'
+        onClick: @togglePreview
```

Comment:
I just remembered there's this `isClickable` thing. Check its usage in `resources/assets/coffee/shared.coffee`